### ISSUE TITLE
In clear(), remove keys in bulk instead of one by one. 

### ISSUE THREAD

**Issue by wlandau** [richfitz/storr#121]
`clear()` removes keys one by one, which is slower than it could be for RDS `storr`s.

https://github.com/richfitz/storr/blob/0c64f1efb0574b46059a962de9d264a74d448843/R/storr.R#L207

``` r
library(storr)
init_storr <- function() {
  s <- storr_rds(tempfile())
  keys <- as.character(seq_len(1e4))
  values <- rep(TRUE, length(keys))
  s$mset(keys, values)
  s
}

s <- init_storr()
system.time(s$clear())
#>    user  system elapsed 
#>   0.244   0.866   1.111

s <- init_storr()
system.time(unlink(file.path(s$path, "keys", s$default_namespace), recursive = TRUE))
#>    user  system elapsed 
#>       0       0       0
```

<sup>Created on 2020-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

@richfitz, would you be open to driver-specific `clear()` methods so we can do whatever is fastest for the situation?

**Comment by richfitz** [comment]
Yes, absolutely. Mdel might be the best method name? Work has been all encompassing sorry with covonavirus this year

**Comment by richfitz** [comment]
Have a look at

```
system.time(s$del(s$list()))
```

this is *still* quite slow. In the special case of dropping a whole namespace we can do better.  So if the driver has a `clear` method that takes a namespace, then the storr object should use it, otherwise it should delete
